src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -43,7 +43,7 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 	addr := hostIncludingPort(req, 443)
 
 	// Establish outbound connection
-	connOut, err := client.getBalancer().DialQOS("tcp", addr, targetQOS(req))
+	connOut, err := client.getBalancer().DialQOS("tcp", addr, client.targetQOS(req))
 	if err != nil {
 		respondBadGateway(clientConn, fmt.Sprintf("Unable to handle CONNECT request: %s", err))
 		return
```

Comment:
Is it possible to get an err back here where connOut is not nil? I.e. is it possible there's a connection closing leak here because we're not calling something like:

```go
if connOut != nil {
    defer connOut.Close()
}
```